keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/account/FixedAccountCredentialResource.java

```diff
@@ -0,0 +1,234 @@
+        auth.require(AccountRoles.MANAGE_ACCOUNT);
+        session.userCredentialManager().updateCredentialLabel(realm, user, credentialId, userLabel);
+    }
+
+    /**
+     * Move a credential to a position behind another credential
+     * @param credentialId The credential to move
+     */
+    @Path("{credentialId}/moveToFirst")
+    @POST
+    public void moveToFirst(final @PathParam("credentialId") String credentialId){
+        moveCredentialAfter(credentialId, null);
+    }
+
+    /**
+     * Move a credential to a position behind another credential
+     * @param credentialId The credential to move
+     * @param newPreviousCredentialId The credential that will be the previous element in the list. If set to null, the moved credential will be the first element in the list.
+     */
+    @Path("{credentialId}/moveAfter/{newPreviousCredentialId}")
+    @POST
+    public void moveCredentialAfter(final @PathParam("credentialId") String credentialId, final @PathParam("newPreviousCredentialId") String newPreviousCredentialId){
+        auth.require(AccountRoles.MANAGE_ACCOUNT);
+        session.userCredentialManager().moveCredentialTo(realm, user, credentialId, newPreviousCredentialId);
+    }
+
+    @GET
+    @Path("password")
+    @Produces(MediaType.APPLICATION_JSON)
+    public PasswordDetails passwordDetails() throws IOException {
+        auth.requireOneOf(AccountRoles.MANAGE_ACCOUNT, AccountRoles.VIEW_PROFILE);
+        
+        PasswordCredentialProvider passwordProvider = (PasswordCredentialProvider) session.getProvider(CredentialProvider.class, PasswordCredentialProviderFactory.PROVIDER_ID);
+        CredentialModel password = passwordProvider.getPassword(realm, user);
+
+        PasswordDetails details = new PasswordDetails();
+        if (password != null) {
+            details.setRegistered(true);
+            details.setLastUpdate(password.getCreatedDate());
+        } else {
+            details.setRegistered(false);
+        }
+
+        return details;
+    }
+
+    @POST
+    @Path("password")
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response passwordUpdate(PasswordUpdate update) {
+        auth.require(AccountRoles.MANAGE_ACCOUNT);
+        
+        event.event(EventType.UPDATE_PASSWORD);
+
+        UserCredentialModel cred = UserCredentialModel.password(update.getCurrentPassword());
+        if (!session.userCredentialManager().isValid(realm, user, cred)) {
+            event.error(org.keycloak.events.Errors.INVALID_USER_CREDENTIALS);
+            return ErrorResponse.error(Messages.INVALID_PASSWORD_EXISTING, Response.Status.BAD_REQUEST);
+        }
+        
+        if (update.getNewPassword() == null) {
+            return ErrorResponse.error(Messages.INVALID_PASSWORD_EXISTING, Response.Status.BAD_REQUEST);
+        }
+        
+        String confirmation = update.getConfirmation();
+        if ((confirmation != null) && !update.getNewPassword().equals(confirmation)) {
```

Comment:
String.equals(...) supports case when parameter is null so we don't need to check for `confirmation!=null`

Reply:
As this is copy-pasted KC code, I'll let it as is

Reply:
Thanks for the information: this is something I'll have to submit in my PR with cleaning of their code ;-)

Reply:
:-)